src/batched/KokkosBatched_Gemm_Decl.hpp

```diff
@@ -332,15 +416,25 @@ int BatchedGemm(const BatchedGemmHandleType *handle, const ScalarType alpha,
 
   switch (handle->get_kernel_algo_type()) {
     case BaseKokkosBatchedAlgos::KK_SERIAL:
-      ret = BatchedSerialGemm<ArgTransA, ArgTransB, Algo::Gemm::Unblocked,
-                              ArgBatchSzDim, ResultsPerThread::Rank2,
-                              ScalarType, AViewType, BViewType, CViewType>(
-                alpha, A, B, beta, C)
-                .invoke();
+      ret =
+          Impl::BatchedSerialGemm<ArgTransA, ArgTransB, Algo::Gemm::Unblocked,
+                                  ArgBatchSzDim, ResultsPerThread::Rank2,
```

Comment:
I might be wrong. Is this interface calling the used-to-be SerialOpt2 algorithm? so should it be ```ResultsPerThread::Rank0```?

Reply:
This is calling the default `SerialGemm` algorithm. "SerialOpt2" has been renamed to `KK_SERIAL_RANK0`.

Reply:
@e10harvey Thanks, I see. But in this file, ```case GemmKokkosBatchedAlgos::KK_SERIAL_RANK0``` is an empty case region. Similar are ```KK_TEAM```, ```KK_TEAMVECTOR```, ```KK_SERIALSIMD```, ```KK_TEAMSIMD```, ```KK_SERIAL_SHMEM```. Are you going to add these calls in this PR or in another PR?

Reply:
@vqd8a: Thanks for your reviews. Adding `KK_SERIAL_RANK0` here is unrelated to #1045; the changes for `KK_SERIAL_RANK0` have just been pushed. Everything except `KK_SERIAL_RANK0`, `KK_SERIAL`, `SQUARE`, and `KK_DBLBUF` will be added in a separate PR.

Reply:
Thanks @e10harvey